### PR TITLE
chore: CI/CD for publishing packages

### PR DESCRIPTION
**Pull request by kamleshFC**

### CODE REVIEW

File: example/todo-app/config/webpack.config.js [hunk 1]
```diff
@@ -1,108 +1,108 @@
-'use strict';
+'use strict'
```
Comment (kamleshFC) [inline example/todo-app/config/webpack.config.js:2]: Mostly format changes. Only check the alias in this file

File: src/hooks/use-rewire-state.ts
Comment (bishalfan) [inline src/hooks/use-rewire-state.ts:1]: we may have to create legacy hooks as well @kamleshFC

File: src/hooks/use-global-state.ts [hunk 2]
```diff
@@ -51,5 +51,7 @@ export const useGlobalState: UseGlobalStateType = function (
       actions.mount?.(null)
     }
   }, [])
-  return [key, state, actions]
+  return useMemo(() => {
```
[the remark this answers is not in the document]
Comment (kamleshFC) [reply src/hooks/use-global-state.ts:54]: Not really, but this is good practise for writing react hook

File: src/hooks/use-rewire-state.ts
Comment (kamleshFC) [reply src/hooks/use-rewire-state.ts:1]: seems to

File: src/helper/key-handler.ts [hunk 3]
```diff
@@ -86,7 +86,7 @@ export const keyHandler = {
   ): T | null {
     for (const eachParentKey in parentsMap) {
       if (keyHandler.checkIfParent(compKey, eachParentKey)) {
-        return parentsMap[eachParentKey]
+        return parentsMap[eachParentKey] ?? null
```
Comment (kamleshFC) [inline src/helper/key-handler.ts:89]: check

File: tsconfig.json [hunk 4]
```diff
@@ -1,35 +1,39 @@
 {
   "compilerOptions": {
+    "esModuleInterop": true,
+    "skipLibCheck": true,
+    "target": "es2022",
+    //    "verbatimModuleSyntax": true,
+    "allowJs": true,
```
Comment (bishalfan) [inline tsconfig.json:7]:
```suggestion
    "allowJs": false,
```

**Review by bishalfan — APPROVED** [review]

**Pull request merged**